js/piwik.js

```diff
@@ -4962,14 +5000,23 @@ if (typeof Piwik !== 'object') {
                 /**
                  * Set heartbeat (in seconds)
                  *
-                 * @param int minimumVisitLength
-                 * @param int heartBeatDelay
+                 * @param int heartBeatDelayInSeconds Defaults to 15.
                  */
-                setHeartBeatTimer: function (minimumVisitLength, heartBeatDelay) {
-                    var now = new Date();
+                enableHeartBeatTimer: function (heartBeatDelayInSeconds) {
+                    configHeartBeatDelayInSeconds = (heartBeatDelayInSeconds || 15) * 1000;
 
-                    configMinimumVisitTime = now.getTime() + minimumVisitLength * 1000;
-                    configHeartBeatTimer = heartBeatDelay * 1000;
+                    // if a tracking request has already been sent, start the heart beat timeout
+                    if (lastTrackerRequestTime !== null) {
```

Comment:
I wonder if we should require eg a minimum timeout of 1 second? Otherwise it could come - in theory - to race conditions. Meaning the heart-beat request could be processed on the server before eg the first pageview which would lead to errors. It is possible eg in a load balanced environment but it is pretty unlikely. Doubt anyone would pass here less than `1` anyway so most likely not needed and would make code only complex